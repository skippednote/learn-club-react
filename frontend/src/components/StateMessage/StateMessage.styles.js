import styled, { css } from 'react-emotion';

const styles = {
  base: css({
    padding: '20px',
    color: 'white',
    'text-align': 'center',
    'margin-top': '10px',
    'margin-bottom': '10px'
  }),
  error: css({
    background: '#e3342f'
  }),
  loading: css({
    background: '#3490dc'
  })
};

export const StateMessage = styled('div')`
  ${styles.base};
  ${({ state }) => styles[state]};
`;
StateMessage.displayName = 'StateMessage';
