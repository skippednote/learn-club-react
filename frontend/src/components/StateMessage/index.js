// @flow
import * as React from 'react';
import { StateMessage as Wrapper } from './StateMessage.styles';

type Props = {
  children: React.Node,
  state: string
};

const StateMessage = ({ children, state }: Props) => (
  <Wrapper state={state}>{children}</Wrapper>
);
export default StateMessage;
