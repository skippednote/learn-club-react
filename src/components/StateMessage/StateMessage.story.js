import React from 'react';
import { storiesOf } from '@storybook/react';
import StateMessage from './index';

storiesOf('StateMessage', module)
  .add('Loading', () => (
    <StateMessage state={'loading'}>
      We are loading the content for you...
    </StateMessage>
  ))
  .add('Error', () => (
    <StateMessage state={'error'}>Oh no!! Failed the request.</StateMessage>
  ));
