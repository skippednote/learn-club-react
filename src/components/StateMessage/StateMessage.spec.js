import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import StateMessage from './index';

describe('StateMessage', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<StateMessage state="error">Error</StateMessage>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders children content', () => {
    const wrapper = shallow(
      <StateMessage state="loading">Loading</StateMessage>
    );
    expect(wrapper.children().text()).toContain('Loading');
  });
});
