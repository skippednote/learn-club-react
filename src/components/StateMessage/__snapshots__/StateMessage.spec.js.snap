// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StateMessage matches snapshot 1`] = `
<StateMessage
  state="error"
>
  Error
</StateMessage>
`;
